package.json

```
@@ -103,7 +103,8 @@
   },
   "lint-staged": {
     "{src,scripts,functions,configs,reporting}/**/*.{js,jsx,ts,tsx,json,css,scss,md}": [
-      "prettier --write"
+      "prettier --write",
+      "eslint src scripts functions --ext .js,.jsx,.ts,.tsx"
```

According to https://github.com/okonet/lint-staged#filtering-files I believe that we probably only need something like:

```suggestion
      "eslint --ext .js,.jsx,.ts,.tsx"
```

or maybe even just:

```suggestion
      "eslint"
```

The glob of files from the initial `lint-staged` selector is passed to the command specified here.

As is, this is still going to let you commit files when there are `eslint` warnings though, which in CI will be treated as errors, and thus require changes to be made to fix it.

If you wanted to emulate this same functionality in this `lint-staged` hook, then I believe you could use `--max-warnings=0`:

- https://eslint.org/docs/user-guide/command-line-interface#options

So combined with the above, would potentially look something like:

```suggestion
      "eslint --max-warnings=0 --ext .js,.jsx,.ts,.tsx"
```

---

2 quick notes on this one.

1. The reason for the `src scripts functions` part is that linting `reporting`, annoyingly enough, includes any `node_modules` directory that is in there. So if we want to remove that, we should also ensure `node_modules` is ignored.

2. The list of file extensions is being supplied because (IIRC) otherwise, eslint tries to run over a wider range of file extensions and finds a variety of things it doesn't understand. I may be wrong though, so give that a try.